normalization/src/main/java/zemberek/normalization/NumberTextConverter.java

```diff
@@ -0,0 +1,206 @@
+package zemberek.normalization;
+
+import zemberek.core.turkish.PrimaryPos;
+import zemberek.core.turkish.SecondaryPos;
+import zemberek.morphology.TurkishMorphology;
+import zemberek.morphology.analysis.SentenceAnalysis;
+import zemberek.morphology.analysis.SingleAnalysis;
+import zemberek.morphology.analysis.WordAnalysis;
+import zemberek.morphology.analysis.tr.TurkishNumbers;
+import zemberek.tokenization.Token;
+import zemberek.tokenization.TurkishTokenizer;
+
+import java.text.DecimalFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class NumberTextConverter {
+
+    private TurkishMorphology morphology;
+    private String patternToFind = "(\\s\\d+\\s(buçuk)\\s)|(\\s(bir|iki|üç|dört|beş|altı|yedi|sekiz|dokuz|on|yirmi|otuz|kırk|elli|atmış|altmış|yetmiş|seksen|doksan|yüz|bin|milyon|milyar)\\s(buçuk)\\s)";
+    private Pattern halfNumberPatternT2N;
+
+    public NumberTextConverter(TurkishMorphology morphology) {
+        this.morphology = morphology;
+        this.halfNumberPatternT2N = Pattern.compile(patternToFind);
```

Comment:
Can be static final defined as a class parameter. No need to initialize it here.